cast/src/starknet_commands/declare.rs

```diff
@@ -14,36 +13,64 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let _ = Command::new("scarb")
+        .current_dir(std::env::current_dir().expect("failed to obtain current dir"))
+        .arg("build")
+        .output()
+        .expect("Failed to build contracts with Scarb");
+
+    let current_dir = std::env::current_dir().expect("failed to obtain current dir");
+    let paths = std::fs::read_dir(format!("{}/target/dev", current_dir.to_str().unwrap()))
+        .expect("failed to read the file maybe build failed");
```

Comment:
what are possible issues here? I can think of a scenario that files does not exist, because a build failed (and probably we should not reach this point in that case). Is there any other case? if not, let's maybe adjust the message accordingly.

Reply:
I encountered weird behaviors with the scarb build command (it should break the execution but sometimes it didn't)., so I thought a hint would be ok here. If you have any specific suggestions, I'd appreciate it if you provided them. 

> what are possible issues here? I can think of a scenario that files does not exist, because a build failed (..). Is there any other case?

I don't see any other case.

Reply:
> it should break the execution but sometimes it didn't
this does not sound good. do you have some idea on what could be the cause? Ideally this is something that should be deterministic, and we should always end execution when such scenario happens.

Reply:
cc @maciektr @mkaput

Reply:
Why are you cc'ing them for? It is for me to confirm/check this again.

Reply:
If you have a case where build fails with exit code 0, please let us know :D

Reply:
> What are you cc'ing them for? It is for me to confirm/check this again.
To make sure we report the bug

Reply:
It's not the scarb team's responsibility that we report bugs to them, still mine, no need to bother them.

Reply:
[fix executing scarb command](https://github.com/software-mansion/protostar/pull/2138/commits/1c5b95973e1dec69b7a55b03f3a8f03a940c18d1)

Reply:
turned out I used Rust's `Command` incorrectly. The `expect` there is for starting a command but we should also check the result's status code.